### PR TITLE
[improve][docs] Use metadataServiceUri in favor of zkServers

### PR DESCRIPTION
**Pull request by nahguam**
Use metadataServiceUri in favor of zkServers

### Motivation

The docs still refer to using the deprecated `zkServers` in the [bare-metal deployment guide](https://pulsar.apache.org/docs/deploy-bare-metal/#deploy-a-bookkeeper-cluster). Its replacement, `metadataServiceUri` is also missing from the [bookkeeper configuration](https://pulsar.apache.org/docs/reference-configuration/#bookkeeper).

### Modifications

* Update the bare-metal deployment guid to use `metadataServiceUri` instead of `zkServers`
* Add `metadataServiceUri` to the bookkeeper configuration
* Mark `zkServers` as deprecated in the bookkeeper configuration
* Fix a mistake in bookkeeper.conf that refers to `zkLedgers` instead of `zkServers`

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: site2/docs/deploy-bare-metal.md [hunk 1]
```diff
@@ -326,15 +326,15 @@ Flag | Description
 
 [BookKeeper](https://bookkeeper.apache.org) handles all persistent data storage in Pulsar. You need to deploy a cluster of BookKeeper bookies to use Pulsar. You can choose to run a **3-bookie BookKeeper cluster**.
 
-You can configure BookKeeper bookies using the [`conf/bookkeeper.conf`](reference-configuration.md#bookkeeper) configuration file. The most important step in configuring bookies for our purposes here is ensuring that [`zkServers`](reference-configuration.md#bookkeeper-zkServers) is set to the connection string for the ZooKeeper cluster. The following is an example:
+You can configure BookKeeper bookies using the [`conf/bookkeeper.conf`](reference-configuration.md#bookkeeper) configuration file. The most important step in configuring bookies for our purposes here is ensuring that `metadataServiceUri` is set to the URI for the ZooKeeper cluster. The following is an example:
 
 ```properties
 
-zkServers=zk1.us-west.example.com:2181,zk2.us-west.example.com:2181,zk3.us-west.example.com:2181
+metadataServiceUri=zk://zk1.us-west.example.com:2181/ledgers
```
Comment (codelipenghui) [inline site2/docs/deploy-bare-metal.md:333]: I think `/ledgers` is not required?

Comment (nahguam) [reply site2/docs/deploy-bare-metal.md:333]: If I do the same connect string as in the `initialize-cluster-metadata` command - `zk:zookeeper:2181` - I get:

```
2022-06-22T10:22:37,902+0000 [main] ERROR org.apache.bookkeeper.server.Main - Failed to build bookie server
org.apache.bookkeeper.bookie.BookieException$MetadataStoreException: Failed to initialize metadata bookie driver
...
Caused by: org.apache.bookkeeper.meta.exceptions.MetadataException: java.lang.IllegalArgumentException: Invalid metadata service URI format: zk:zookeeper:2181
...
Caused by: java.lang.IllegalArgumentException: Invalid metadata service URI format: zk:zookeeper:2181
...
```

If I use a URI but omit the `/ledgers` path - `zk://zookeeper:2181` - I get:

```
[main] ERROR org.apache.bookkeeper.server.Main - Failed to build bookie server
java.lang.IllegalArgumentException: Path length must be > 0
...
```

Only `zk://zookeeper:2181/ledgers` is working for me

**Review by codelipenghui — APPROVED** [review]

File: conf/bookkeeper.conf [hunk 2]
```diff
@@ -631,10 +631,10 @@ diskCheckInterval=10000
 ## Metadata Service settings
 #############################################################################
 
-# metadata service uri that bookkeeper is used for loading corresponding metadata driver and resolving its metadata service location
+# Metadata service uri that bookkeeper uses for loading the corresponding metadata driver and resolving its metadata service location
```
Comment (Anonymitaet) [inline conf/bookkeeper.conf:634]:
```suggestion
# Metadata service URI that bookkeeper uses for loading the corresponding metadata driver and resolving its metadata service location
```
Capitalize proper nouns correctly. https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.je5thtkj42cb

```suggestion
# Metadata service uri that bookkeeper uses for loading the corresponding metadata driver and resolving its metadata service location
```

File: conf/bookkeeper.conf [hunk 3]
```diff
@@ -631,10 +631,10 @@ diskCheckInterval=10000
...
 # Examples: 
-#  - metadataServiceUri=metadata-store:zk:my-zk-1:2181
-#  - metadataServiceUri=metadata-store:etcd:http://my-etcd:2379
+#  - metadataServiceUri=zk://my-zk-1:2181/ledgers
+#  - metadataServiceUri=etcd:http://my-etcd:2379
```
Comment (Anonymitaet) [inline conf/bookkeeper.conf:637]: FYI @momo-jun @daveduggins

File: site2/docs/reference-configuration.md [hunk 4]
```diff
@@ -72,7 +72,8 @@ BookKeeper is a replicated log storage system that Pulsar uses for durable stora
 |flushInterval|How long the interval to flush ledger index pages to disk, in milliseconds. Flushing index files will introduce much random disk I/O. If separating journal dir and ledger dirs each on different devices, flushing would not affect performance. But if putting journal dir and ledger dirs on same device, performance degrade significantly on too frequent flushing. You can consider increment flush interval to get better performance, but you need to pay more time on bookie server restart after failure.|60000|
 |bookieDeathWatchInterval|Interval to watch whether bookie is dead or not, in milliseconds|1000|
 |allowStorageExpansion|Allow the bookie storage to expand. Newly added ledger and index dirs must be empty.|false|
-|zkServers|A list of one of more servers on which zookeeper is running. The server list can be comma separated values, for example: zkServers=zk1:2181,zk2:2181,zk3:2181.|localhost:2181|
+|metadataServiceUri|Metadata service uri that bookkeeper uses for loading the corresponding metadata driver and resolving its metadata service location.||
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:75]:
```suggestion
|metadataServiceUri|Metadata service URI that BookKeeper uses for loading the corresponding metadata driver and resolving its metadata service location.||
```
```suggestion
|metadataServiceUri|Metadata service uri that bookkeeper uses for loading the corresponding metadata driver and resolving its metadata service location.||
```

File: site2/docs/reference-configuration.md [hunk 5]
```diff
@@ -72,7 +72,8 @@ BookKeeper is a replicated log storage system that Pulsar uses for durable stora
...
+|zkServers|Deprecated in favor of using `metadataServiceUri`. A list of one of more servers on which zookeeper is running. The server list can be comma separated values, for example: zkServers=zk1:2181,zk2:2181,zk3:2181.|localhost:2181|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:76]:
```suggestion
|zkServers|Deprecated in favor of using `metadataServiceUri`. A list of one or more servers on which ZooKeeper is running. The server list can be comma-separated values, for example: zkServers=zk1:2181,zk2:2181,zk3:2181.|localhost:2181|
```
```suggestion
|zkServers|Deprecated in favor of using `metadataServiceUri`. A list of one of more servers on which zookeeper is running. The server list can be comma separated values, for example: zkServers=zk1:2181,zk2:2181,zk3:2181.|localhost:2181|
```

**Comment by Raven888888** [comment]
For multiple zk, is it comma separated like this?
`metadataServiceUri=zk://my-zk-1:2181/ledgers,zk://my-zk-2:2181/ledgers`

**Comment by nahguam** [comment]
Hi @DaveDuggins Does this look ok?

File: conf/bookkeeper.conf (hunk 3 above)
Comment (DaveDuggins) [reply conf/bookkeeper.conf:637]: > Hi @DaveDuggins Does this look ok

Yes. Anonymitaet pinged me for reference. I'm not the approver. We still need a committer to merge this.

**Comment by momo-jun** [comment]
@nahguam can you pls resolve the conflicting files?

**Comment by nahguam** [comment]
> @nahguam can you pls resolve the conflicting files?

@momo-jun Done

**Comment by DaveDuggins** [comment]
I'm working this now.  Thank you Sijie!

File: conf/bookkeeper.conf [hunk 6]
```diff
@@ -642,10 +642,10 @@ diskCheckInterval=10000
...
```
Comment (tisonkun) [inline conf/bookkeeper.conf:648]: I'm not quite sure whether this config pattern is correct.

I'm using the `metadata-store:` variant in prod env.

cc @merlimat @eolivelli

**Review by tisonkun — APPROVED** [review]
LGTM. Pending to merge...

**Comment by tisonkun** [comment]
Merging...

**Comment by tisonkun** [comment]
Thank you!

**Pull request merged**